### PR TITLE
Fix #2063: Modify j.u.Date#toString to be Java 8 compliant.

### LINKED ISSUE #2063: j.u.Date#toString output does not match  JVM practice.
Later: fixed by PR #2064 

Presenting problem: 

Current behavior of DateTest.scala Date(1490986064740)  
expected JVM string: "Fri Mar 31 14:47:44 EDT 2017" 

This appears to have been an implementation felicity.  The fix should be non-controversial.

### PR DESCRIPTION
**Pull request by LeeTibbert**
We modify the toString method to match the results of at least Java 8
and earlier.

### CODE REVIEW

**Comment by LeeTibbert** [comment]
I believe this PR is ready for review.

PR #2160 contains fixes required by this PR. The former has now been merged, so 
I re-based this PR and ran the CI tests again.

All tests except `check-lint` pass. The `check-lint` failure appears to be
a bug which crept into the CI configuration approximately yesterday, 2021-02-24.
I changed no files here, and scalafmt/check-lint passed a month or so ago,
so I believe that this PR will merge cleanly once the unrelated `check-lint`
but is fixed.

File: javalib/src/main/scala/java/util/Date.scala [hunk 1]
```diff
@@ -35,12 +40,44 @@ class Date(var milliseconds: Long)
...
 
 object Date {
+  // Provide prerequisite for localtime_r calls.
+  // Applications which must track timezone changes over their lifetime
+  // must do timely subsequent tzset() calls, either directly or through
+  // an occasional localtime().
+
+  tzset()
+
+  private def secondsToString(seconds: Long, default: String): String = Zone {
```
Comment (WojciechMazur) [inline javalib/src/main/scala/java/util/Date.scala:60]:
```suggestion
  private def secondsToString(seconds: Long, default: => String): String = Zone {
```
Let's make default lazy by using by-name parameter, so we don't need to allocate it if it's not needed

File: javalib/src/main/scala/java/util/Date.scala [hunk 2]
```diff
@@ -35,12 +40,44 @@ class Date(var milliseconds: Long)
   def setTime(time: Long): Unit =
     milliseconds = time
 
-  override def toString(): String = s"Date($milliseconds)"
-
   def toInstant(): Instant = Instant.ofEpochMilli(getTime())
+
+  override def toString(): String = {
+    val seconds = milliseconds / 1000L
+    val default = s"Date($milliseconds)"
```
Comment (WojciechMazur) [inline javalib/src/main/scala/java/util/Date.scala:47]:
```suggestion
    def default = s"Date($milliseconds)"
```
To make a lazy evaluation in secondsToString work.

**Review by WojciechMazur — CHANGES_REQUESTED** [review]
Looks good, but we should try not to allocate the default string each time.

File: javalib/src/main/scala/java/util/Date.scala (hunk 1 above)
Comment (LeeTibbert) [reply javalib/src/main/scala/java/util/Date.scala:60]: Thank you for the improvement.  Every nanosecond counts.

**Review by WojciechMazur — APPROVED** [review]

**Pull request merged**